### PR TITLE
mgmt, compute support withplan for VMSS

### PR DESCRIPTION
**Pull request by haolingdong-msft**
# Description

Add feature, issue link: https://github.com/azure/azure-sdk-for-java/issues/24301 

## Code Change
1. Update `VirtualMachineScaleSet` and `VirtualMachineScaleSetImpl` to add withPlan() interface and plan() getter
2. Add test to test creating virtual machine scale set with plan, the LIVE test passed
3. Record the test and add session records.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 1]
```diff
@@ -84,6 +85,61 @@ protected void cleanUpResources() {
...
+                .withExistingResourceGroup(resourceGroup)
+                .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_A0)
+                .withExistingPrimaryNetworkSubnet(network, "subnet1")
+                .withoutPrimaryInternetFacingLoadBalancer()
+                .withoutPrimaryInternalLoadBalancer()
+                .withSpecificLinuxImageVersion(imageReference)
+                .withRootUsername(uname)
+                .withSsh(sshPublicKey())
+                .withUnmanagedDisks()
+                .withNewStorageAccount(generateRandomResourceName("stg", 15))
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:131]: Avoid using unmanaged disk.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 2]
```diff
@@ -84,6 +85,61 @@ protected void cleanUpResources() {
...
+
+        VirtualMachineScaleSet virtualMachineScaleSet =
+            this
+                .computeManager
+                .virtualMachineScaleSets()
+                .define(vmssName)
+                .withRegion(region)
+                .withExistingResourceGroup(resourceGroup)
+                .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_A0)
+                .withExistingPrimaryNetworkSubnet(network, "subnet1")
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:124]: Usually it is easier to call `withNewNetwork`?

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineScaleSet.java [hunk 3]
```diff
@@ -382,6 +382,11 @@ RunCommandResult runShellScriptInVMInstance(
      */
     AdditionalCapabilities additionalCapabilities();
 
+    /**
+     * @return the plan value
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineScaleSet.java:386]: Do we have a better javadoc? `the plan value` is just useless...

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java (hunk 2 above)
Comment (haolingdong-msft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:124]: I refer to other tests to add the network configuration part, will try withNewNetWork(). Thanks

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 4]
```diff
@@ -127,7 +127,7 @@ public void canCreateVMSSWithPlan() {
                 .withSpecificLinuxImageVersion(imageReference)
                 .withRootUsername(uname)
                 .withSsh(sshPublicKey())
-                .withUnmanagedDisks()
+                .withNewDataDisk(1)
                 .withNewStorageAccount(generateRandomResourceName("stg", 15))
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:131]: I don't think you need the storage account?

**Pull request merged**